core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2076,29 +2076,34 @@ class KafkaApis(val requestChannel: RequestChannel,
       // When this callback is triggered, the remote API call has completed
       request.apiRemoteCompleteTimeNanos = time.nanoseconds
     }
-    val quotaSensors = quotas.request.getOrCreateQuotaSensors(request.session.sanitizedUser, clientId)
-    def recordNetworkThreadTimeNanos(timeNanos: Long) {
-      quotas.request.recordNoThrottle(quotaSensors, nanosToPercentage(timeNanos))
-    }
-    request.recordNetworkThreadTimeCallback = Some(recordNetworkThreadTimeNanos)
+    if (quotas.request.quotasEnabled) {
```

Comment:
@ijuma Thank you for the review. I have updated the code. Can you take another look? Thanks.